src/Pact/Native.hs

```diff
@@ -1036,3 +1039,40 @@ baseStrToInt base t =
          else Left $ "character '" <> T.singleton c' <>
                 "' is out of range for base " <> tShow base <> ": " <> t
 {-# INLINABLE baseStrToInt #-}
+
+base64Encode :: NativeDef
+base64Encode = defRNative "base64-encode" go
+  (funType tTyString [("string", tTyString)])
+  ["(base64-encode \"hello world!\")"]
+  "Encode STRING as unpadded base64"
+  where
+    go :: RNativeFun e
+    go i as = case as of
+      [TLitString s] -> return
+        . tStr
+        . T.dropWhileEnd (== '=')
+        . T.decodeUtf8
+        . B64U.encode
+        . T.encodeUtf8
+        $ s
+      _ -> argsError i as
+
+base64decode :: NativeDef
+base64decode = defRNative "base64-decode" go
+  (funType tTyString [("string", tTyString)])
+  ["(base64-decode \"aGVsbG8gd29ybGQh\")"]
+  "Decode STRING from unpadded base64"
+  where
+    go :: RNativeFun e
+    go i as = case as of
+      [TLitString s] ->
+        case B64U.decode . T.encodeUtf8 $ pad s of
+          Left e -> evalError' i
+            $ "Could not decode string: "
+            <> pretty e
+          Right t -> return . tStr $ T.decodeUtf8 t
+      _ -> argsError i as
+
+    pad t =
+      let s = T.length t `mod` 4
```

Comment:
Isn't this code replicated somewhere else?

Reply:
I've moved this to `Pact.Utils` so it's only written once.